GameFighter/Models/Army.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+
+namespace GameFighter.Models
+{
+    public class Army
+    {
+        public readonly List<Warrior> ArmyMembers = new List<Warrior>();
+
+        public int CountUnits => ArmyMembers.Count;
+
+        public bool HasAliveUnit
+        {
+            get
+            {
+                foreach (var item in ArmyMembers)
+                {
+                    if (item.IsAlive)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        public void AddUnits(Warrior unit, int number)
```

Comment:
I still don't understand how you create the units. I see that you create a new object of required type and then calls the method to create one more object of this type.
Read about generic type parameters and constraints